internal/backend/connections/connections.go

```diff
@@ -16,14 +18,28 @@ import (
 type Connections struct {
 	fx.In
 
-	Z  *zap.Logger
-	DB db.DB
+	Z            *zap.Logger
+	DB           db.DB
+	Integrations sdkservices.Integrations
 }
 
 func New(c Connections) sdkservices.Connections { return &c }
 
 func (c *Connections) Create(ctx context.Context, conn sdktypes.Connection) (sdktypes.ConnectionID, error) {
-	conn = conn.WithNewID()
+	intg, err := c.Integrations.GetByID(ctx, conn.IntegrationID())
+	if err != nil {
+		return sdktypes.InvalidConnectionID, err
+	}
+
+	status := intg.Get().InitialConnectionStatus()
+	if !status.IsValid() {
+		// get the connection status of a new connection.
+		if status, err = intg.GetConnectionStatus(ctx, sdktypes.InvalidConnectionID); err != nil {
```

Comment:
that feels very weird get status of invalid

Reply:
you have a weird CTO. 

seriously though - the connection id here is invalid, meaning no connection id.